.github/workflows/build-crates-individually.yml

```diff
@@ -0,0 +1,103 @@
+name: Build crates individually
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      # production code and test code
+      - '**/*.rs'
+      # dependencies
+      - '**/Cargo.toml'
+      - '**/Cargo.lock'
+      # workflow definitions
+      - '.github/workflows/build-crates-individually.yml'
+  pull_request:
+    paths:
+      # production code and test code
+      - '**/*.rs'
+      # dependencies
+      - '**/Cargo.toml'
+      - '**/Cargo.lock'
+      # workflow definitions
+      - '.github/workflows/build-crates-individually.yml'
+
+env:
+  CARGO_INCREMENTAL: 0
+  RUST_BACKTRACE: full
+  RUST_LIB_BACKTRACE: full
+  COLORBT_SHOW_HIDDEN: '1'
+
+jobs:
+  matrix:
+    name: Crates matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3.0.2
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - uses: actions-rs/cargo@v1.0.3
+      - id: set-matrix
+        name: Dinamically build crates JSON
+        run: |
+          TEMP_DIR=$(mktemp -d)
+          echo "$( cargo tree --depth 0 --edges no-normal,no-dev,no-build,no-proc-macro --format {p} --prefix none | cut -d ' ' -f1 | sed '/^$/d' | awk '{ printf "\"%s\",\n", $0 }' | sed '$ s/.$//')" > $TEMP_DIR/crates.txt
+          JSON_CRATES=$(cat $TEMP_DIR/crates.txt)
+          MATRIX=$((
+            echo '{ "crate" : ['
+            echo "${JSON_CRATES}"
+            echo " ]}"
+          ) | jq -c .)
+          echo $MATRIX
+          echo $MATRIX | jq .
+          echo "::set-output name=matrix::$MATRIX"
+
+  check-matrix:
+    runs-on: ubuntu-latest
+    needs: [ matrix ]
+    steps:
+      - name: Install json2yaml
+        run: |
+          sudo npm install -g json2yaml
+
+      - name: Check matrix definition
+        run: |
+          matrix='${{ needs.matrix.outputs.matrix }}'
+          echo $matrix
+          echo $matrix | jq .
+          echo $matrix | json2yaml
+
+  build:
+    name: Build ${{ matrix.crates.crate }} crate
+    needs: [ matrix, check-matrix ]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        features: ['', '--all-features', '--no-default-features']
```

Comment:
It might be faster to run these commands sequentially, so they can re-use the local cargo cache. Some Zebra crates do not have any features, and most don't have any default features.

The most efficient order starts with the smallest number of features:
```sh
cargo build --no-default-features
cargo build
cargo build --all-features
```

Reply:
Fixed by: [c14c51d](https://github.com/ZcashFoundation/zebra/pull/4640/commits/c14c51d1e3a7e76208d9d1594bd81774e46e0eb2)